sw/device/lib/base/macros.h

```diff
@@ -418,4 +418,17 @@
   if (expr_) {            \
   }
 
+/**
+ * Stub for the HW top header file.
+ * The definition can point to a different path with a compile flag.
+ */
+#ifndef OPENTITAN_HW_TOP_HDR_PATH
+#define OPENTITAN_HW_TOP_HDR_PATH "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#endif
+
+#ifndef OPENTITAN_HW_TOP_MEMORY_HDR_PATH
+#define OPENTITAN_HW_TOP_MEMORY_HDR_PATH \
+  "hw/top_earlgrey/sw/autogen/top_earlgrey_memory.h"
+#endif
+
```

Comment:
IIUC we want to support multiple tops in a single repo, and assuming no source code change (might not be possible for every module), we would need multiple targets for most if not all libraries.

If that's the case we should probably look into defining this default somewhere in the build system and introduce a macro to handle target names and dependencies for multiple tops, e.g. `toplevels = ["earlgrey", "somehing_else"] ` @cfrantz wdyt?

Reply:
In bazel, it is possible to use config to define such properties. However, you still need in the source code to link the bazel config with the stand-ins. In this PR, the stand-in macros are the two defined in `macros.h`. 

Regarding the location, I figured `sw/devices/lib/base/macros.h` would be appropriate because those are a device-specific macros. Where as for bazel, such group of comfigs kind of exists in https://github.com/lowRISC/opentitan/blob/master/sw/device/tock/BUILD. We probably need to move it elsewhere though.